libcontainer/rootfs_linux.go

```diff
@@ -53,7 +53,7 @@ func prepareRootfs(pipe io.ReadWriter, iConfig *initConfig) (err error) {
 			}
 		}
 
-		if err := mountToRootfs(m, config.Rootfs, config.MountLabel); err != nil {
+		if err := mountToRootfs(m, config.Rootfs, config.MountLabel, config.Namespaces.Contains(configs.NEWNS)); err != nil {
```

Comment:
Why are you using NEWNS here when the description of the PR talks about NEWPID?

Reply:
sorry that was a leftover of a cleanup.  The patch was originally written to address only the "chroot" case but I've changed it to address also another problem: `/proc` could not be bind mounted from the host.  This is useful for rootless containers when the PID namespace is shared with the host (and the container cannot mount a new `/proc`).

I've pushed a new version which is much simpler now.